File: backend/src/main/java/org/luxons/sevenwonders/actions/JoinGameAction.java
package org.luxons.sevenwonders.actions;

import javax.validation.constraints.NotNull;

import org.hildan.livedoc.core.annotations.types.ApiType;
import org.luxons.sevenwonders.doc.Documentation;

/**
 * The action to join a game.
 */
@ApiType(group = Documentation.GROUP_ACTIONS)
public class JoinGameAction {

    @NotNull
    private Long gameId;

    /**
     * @return The ID of the game to join
     */
    public Long getGameId() {
        return gameId;
    }

    public void setGameId(Long gameId) {
        this.gameId = gameId;
    }
}

File: backend/src/main/kotlin/org/luxons/sevenwonders/actions/JoinGameAction.kt
package org.luxons.sevenwonders.actions

import org.hildan.livedoc.core.annotations.types.ApiType
import org.hildan.livedoc.core.annotations.types.ApiTypeProperty
import org.luxons.sevenwonders.doc.Documentation

/**
 * The action to join a game.
 */
@ApiType(group = Documentation.GROUP_ACTIONS)
class JoinGameAction(
    /**
     * The ID of the game to join.
     */
    @ApiTypeProperty(required = true)
    val gameId: Long
)
